drivers/sensor/mpu9250/ak8963.c

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright (c) 2021, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/i2c.h>
+#include <drivers/sensor.h>
+
+#include "mpu9250.h"
+#include "ak8963.h"
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(MPU9250, CONFIG_SENSOR_LOG_LEVEL);
+
+
+int ak8963_convert_magn(struct sensor_value *val, int16_t raw_val,
+			 int16_t scale, uint8_t st2)
+{
+	/* The sensor device returns 10^-9 Teslas after scaling.
+	 * Scale adjusts for calibration data and units
+	 * So sensor instance returns Gauss units
+	 */
+
+	/* If overflow happens then value is invalid */
+	if ((st2 & AK8963_ST2_OVRFL_BIT) != 0) {
+		LOG_INF("Magnetometer value overflow.");
+		return -EOVERFLOW;
+	}
+
+	int64_t scaled_val = (int64_t)raw_val * (int64_t)scale;
```

Comment:
the result of int16 * int16 fits into int32.

Reply:
Chnaged